src/routes/v1/error.rs

```diff
@@ -0,0 +1,66 @@
+use actix_web::{HttpResponse, ResponseError};
+use reqwest::StatusCode;
+use serde::Serialize;
+use thiserror::Error;
+
+#[derive(Serialize)]
+pub struct ErrorResponse {
+    pub code: u16,
+    pub error: String,
+    pub message: String,
+}
+
+#[derive(Error, Debug)]
+pub enum CommonError {
+    #[error("Not found")]
+    NotFound,
+    #[error("Bad request {0}")]
+    BadRequest(String),
+    #[error("Requested user was not found")]
+    UserNotFound,
+    #[error("{0}")]
+    Forbidden(String),
+    #[error("Unknown Internal Error")]
+    Unknown,
+    #[error("Unauthorized")]
+    Unauthorized,
+    #[error("Too many requests")]
+    TooManyRequests,
+}
+
+impl CommonError {
+    pub fn name(&self) -> String {
+        match self {
+            Self::NotFound => "Not found".to_string(),
+            Self::BadRequest(_str) => "Bad request".to_string(),
+            Self::UserNotFound => "UserNotFound".to_string(),
+            Self::Unauthorized => "Unauthorized".to_string(),
+            Self::TooManyRequests => "TooManyRequests".to_string(),
+            Self::Forbidden(_str) => "Forbidden".to_string(),
+            Self::Unknown => "Unknown".to_string(),
+        }
```

Comment:
This could also turn to:
```suggestion
        format!("{:?}", self)
```